### PR TITLE
CAMEL-15056: camel-main: provide a way to bring a custom engine to control shutdown logic

### PR DESCRIPTION
**Pull request by lburgazzoli**
[ ] Make sure there is a [JIRA issue](https://issues.apache.org/jira/browse/CAMEL) filed for the change (usually before you start working on it).  Trivial changes like typos do not require a JIRA issue.  Your pull request should address just this issue, without pulling in other changes.
[ ] Each commit in the pull request should have a meaningful subject line and body.
[ ] If you're unsure, you can format the pull request title like `[CAMEL-XXX] Fixes bug in camel-file component`, where you replace `CAMEL-XXX` with the appropriate JIRA issue.
[ ] Write a pull request description that is detailed enough to understand what the pull request does, how, and why.
[ ] Run `mvn clean install -Psourcecheck` in your module with source check enabled to make sure basic checks pass and there are no checkstyle violations. A more thorough check will be performed on your pull request automatically.
Below are the contribution guidelines:
https://github.com/apache/camel/blob/master/CONTRIBUTING.md

### CODE REVIEW

**Comment by lburgazzoli** [comment]
@davsclaus here a small POC to provide a custom shut-down strategy, mind having a look ?

File: core/camel-main/src/main/java/org/apache/camel/main/ShutdownStrategy.java [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.camel.main;
+
+import java.util.concurrent.TimeUnit;
+
+public interface ShutdownStrategy {
```
Comment (davsclaus) [inline core/camel-main/src/main/java/org/apache/camel/main/ShutdownStrategy.java:21]: We already have a org.apache.camel.spi.ShutdownStrategy in camel-api. So lets rename this to something not clashing, maybe `MainShutdownStrategy` ?

Comment (lburgazzoli) [reply core/camel-main/src/main/java/org/apache/camel/main/ShutdownStrategy.java]: should be fixed

**Comment by davsclaus** [comment]
After merging this PR we may want to compile camel-example and camel-spring-boot in case something is affected.

**Comment by lburgazzoli** [comment]
> After merging this PR we may want to compile camel-example and camel-spring-boot in case something is affected.

I'm on it

**Comment by lburgazzoli** [comment]
@davsclaus the spring boot side is also done https://github.com/apache/camel-spring-boot/pull/69

**Pull request merged**